src/Shared/FileUtilities.cs

```diff
@@ -532,10 +532,9 @@ internal static string GetDirectory(string fileSpec)
         /// <returns></returns>
         internal static bool HasExtension(string fileName, string[] allowedExtensions)
         {
-            string fileExtension = Path.GetExtension(fileName);
             foreach (string extension in allowedExtensions)
             {
-                if (String.Compare(fileExtension, extension, StringComparison.CurrentCultureIgnoreCase) == 0)
+                if (fileName.EndsWith(extension, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should this also confirm that `fileName[0] == '.'` ?

Reply:
Path.GetExtension(".exe") returns ".exe" - what would your check cover?

I'll split it into a separate commit, the comparison changed because you don't want this `HasExtension("foo.ini", ".Ini")` to return false when you are running under `tr-TR`.

Reply:
> Path.GetExtension(".exe") returns ".exe" - what would your check cover?

Oops, I meant `extension[0] == '.'`.